drivers/axi_core/axi_dmac/axi_dmac.c

```diff
@@ -90,8 +90,7 @@ void axi_dmac_default_isr(void *instance)
 		axi_dmac_write(dmac, AXI_DMAC_REG_Y_LENGTH, 0x0);
 
 		axi_dmac_write(dmac, AXI_DMAC_REG_START_TRANSFER, 0x1);
-	}
-	if (reg_val & AXI_DMAC_IRQ_EOT) {
+	} else if (reg_val & AXI_DMAC_IRQ_EOT) {
```

Comment:
```
if (dmac->big_transfer.size) {
    if ((reg_val & AXI_DMAC_IRQ_SOT) || (reg_val & AXI_DMAC_IRQ_EOT))
    {
       //Queue new transfer
       big_transfer.size -= new_transfer_size;
    }
    //else other irq, ignore it
} else if (reg_val & AXI_DMAC_IRQ_EOT) {
   dmac->big_transfer.size_done = 0;
   //Other cleanup
}
//else Other irelevant irq
```
What about this?  (It will solve the last point of the issue (https://github.com/analogdevicesinc/no-OS/issues/1074)
I'am not sure if the current approach will do.

Reply:
I updated a little my code suggestion.